exercises/drill-enum1-replace-nils.livemd

```diff
@@ -0,0 +1,163 @@
+# Drill: Replace `nil`s in a List with Elements from a 2nd List
+
+```elixir
+Mix.install([
+  {:youtube, github: "brooklinjazz/youtube"},
+  {:hidden_cell, github: "brooklinjazz/hidden_cell"},
+  {:tested_cell, github: "brooklinjazz/tested_cell"},
+  {:utils, path: "#{__DIR__}/../utils"}
+])
+```
+
+## Navigation
+
+[Return Home](../start.livemd)<span style="padding: 0 30px"></span>
+[Report An Issue](https://github.com/DockYard-Academy/beta_curriculum/issues/new?assignees=&labels=&template=issue.md&title=)
+
+## Purpose
+
+This is a *drill* exercise:
+
+> *Drill* exercises are meant to provide practise of Elixir's
+> syntax and important language modules so developers can
+> type them out as fast as possible.
+> The problem should not be conceptually difficult to
+> facilitate this goal.
+> 
+> Developers need to commit both Elixir's syntax and core
+> modules/functions to (working) memory so that when writing
+> their code the developer is not slowed down having to
+> remember the syntax or which module function to use. This
+> is analogous to knowing how to spell and the grammar of
+> your (human) language.
+
+In this particular iteration of this problem, the goal is
+to practise writing/using functions from the `Enum` module;
+some important ones being:
+
+* Iteration: `map`, `filter`, `reduce`, `zip`
```

Comment:
There's a script in the codebase that will automatically link built-in Elixir code to documentation.

If you write this as `Enum.map/2` for example, it will be replaced with a markdown link to the docs.

Reply:
i didnt mean for there to be links, would you like or prefer them to be?

Reply:
I find it useful for students to have links to see the docs. The nice thing is when you run the `mix bc` script it will automatically find code like `Enum.map/2` and replace it with a link to the documentation.